Inline Assembly Error in helpers.c

Hi!

When building a Rust project which uses wasmtime::Engine to run WASM binaries, linking fails. Project is built as usual with cargo. The following error occurs in [runtime/src/helpers.c:78](https://github.com/bytecodealliance/wasmtime/blob/main/crates/runtime/src/helpers.c#L78): _undefined reference to `asm'_

To me, it looks like this should be an empty inline assembly statement, and effectively, replacing it with `__asm("");` solves the issue. 

I am unaware of the build process for helpers.c but happy to provide more details if required. OS: Linux 6.2.9, CC: GCC 12.2.1 20230201

--- a/crates/runtime/src/helpers.c
+++ b/crates/runtime/src/helpers.c
@@ -75,7 +75,7 @@ __attribute__((weak, noinline))
 #endif
 void __jit_debug_register_code() {
 #ifndef CFG_TARGET_OS_windows
-  asm("");
+  __asm__("");
 #endif
 }
 
@@ -88,7 +88,7 @@ struct JITDescriptor {
 
 #ifdef CFG_TARGET_OS_windows
   // export required for external access.
-  __declspec(dllexport) 
+  __declspec(dllexport)
 #else
   // Note the `weak` linkage here which is the same purpose as above. We want to
   // let other runtimes be able to override this since our own definition isn't

